kymatio/scattering2d/backend/torch_backend.py

```diff
@@ -70,19 +66,23 @@ def __call__(self, x):
         batch_shape = x.shape[:-2]
         signal_shape = x.shape[-2:]
         x = x.reshape((-1, 1) + signal_shape)
-        if not self.pre_pad:
-            x = self.padding_module(x)
+        x = self.padding_module(x)
 
-            # Note: PyTorch is not effective to pad signals of size N-1 with N
-            # elements, thus we had to add this fix.
-            if self.pad_size[0] == self.input_size[0]:
-                x = torch.cat([x[:, :, 1, :].unsqueeze(2), x, x[:, :, x.shape[2] - 2, :].unsqueeze(2)], 2)
-            if self.pad_size[2] == self.input_size[1]:
-                x = torch.cat([x[:, :, :, 1].unsqueeze(3), x, x[:, :, :, x.shape[3] - 2].unsqueeze(3)], 3)
+        # Note: PyTorch is not effective to pad signals of size N-1 with N
+        # elements, thus we had to add this fix.
+        if self.pad_size[0] == self.input_size[0]:
+            x = torch.cat([x[:, :, 1, :].unsqueeze(2), x, x[:, :, x.shape[2] - 2, :].unsqueeze(2)], 2)
```

Comment:
could remove some unsqueeze operations here

Reply:
what would be the smartest way of doing it? atm im thinking `x[:, :, 1, :].reshape(x.shape[:2] + (1,) + x.shape[3:])`

Reply:
`torch.cat([x[:, :, 1:2], x, x[:, :, -2:-1]])` looks like it should do it. Please check

Reply:
Not in scope for this PR.